src/index.js

```diff
@@ -1,28 +1,79 @@
-import React from 'react';
+import React, { Component } from 'react';
 import ReactDOM from 'react-dom';
+import createHistory from 'history/createBrowserHistory'
 import {
   BrowserRouter as Router,
-  Route
+  Route,
+  matchPath
 } from 'react-router-dom'
 
-import App from './App';
+import Game from './components/Game';
 import Landing from './components/Landing';
 import './index.css';
 
-ReactDOM.render(
-  <Router>
-    <div className='app'>
-      <h1>Album Hangman</h1>
-      <div className='container'>
-        <Route exact path={'/'} component={Landing} />
-        <Route path={'/game/:username/:period?/:hideArtwork?'} component={App} />
+const GAME_PATH = '/game/:username?/:period?/:hideArtwork?';
+
+class App extends Component{
+  constructor(props){
+    super(props);
+
+    // Match current path to keep username and period in state
+    const history = createHistory();
```

Comment:
This, for example, can be avoided if the `App` would be in a separate file.

Reply:
How?

Reply:
Using `match` as before, which is more clear IMO

Reply:
`match` is being passed by the `route` component, so as long as our `App` component isn't rendered from a `route`. Moving `App` to `App.js` will still make us build the match ourselves, unfortunately.